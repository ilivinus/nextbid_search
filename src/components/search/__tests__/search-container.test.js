import React from 'react';
import renderer from 'react-test-renderer';
import SearchContainer from '../search-container';

describe("#SearchContainer",()=>{

    it("SearchContainer doesn't change on keydown",()=>{
        const component = renderer.create(
            <SearchContainer />
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        let props = tree.children[1].children[0].children[0].children[0].props;
        //manually trigger the callback
        props.onKeyDown({ keyCode : 32});
        
        
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});