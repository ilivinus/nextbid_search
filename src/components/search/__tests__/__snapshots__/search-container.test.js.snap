// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SearchContainer SearchContainer doesn't change on keydown 1`] = `
<div
  className="search-container"
>
  <p
    className="search-header"
  >
    <img
      alt="TheNextBid"
      className="logo"
      src="https://thenextbid.com/eeb9b1a373fc912868daaa578bb7ccc0.svg"
    />
    <strong>
      Search Engine
    </strong>
  </p>
  <form
    autoComplete="off"
  >
    <div
      className="suggestion"
    >
      <div
        className="suggestion"
      >
        <input
          className=""
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          className="sb-button"
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        className="suggestion-items"
      />
    </div>
  </form>
  <div />
</div>
`;

exports[`#SearchContainer SearchContainer doesn't change on keydown 2`] = `
<div
  className="search-container"
>
  <p
    className="search-header"
  >
    <img
      alt="TheNextBid"
      className="logo"
      src="https://thenextbid.com/eeb9b1a373fc912868daaa578bb7ccc0.svg"
    />
    <strong>
      Search Engine
    </strong>
  </p>
  <form
    autoComplete="off"
  >
    <div
      className="suggestion"
    >
      <div
        className="suggestion"
      >
        <input
          className=""
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          className="sb-button"
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        className="suggestion-items"
      />
    </div>
  </form>
  <div />
</div>
`;
